views/partials/_mobile-navigation.njk

```diff
@@ -2,11 +2,11 @@
   <ul class="app-mobile-nav__list">
     {% for item in navigation %}
     <li>
-      <a class="govuk-link js-mobile-nav-subnav-toggler" href="/{{ item.url }}">{{ item.label }}</a>
+      <a class="govuk-link js-mobile-nav-subnav-toggler" href="/{{ item.url }}/">{{ item.label }}</a>
```

Comment:
I am _guessing_ that this change (and the other trailing slash changes) is attempting to fix an issue with resolving trailing-slash-less paths when falling back to the service worker's cache?

I don't think we can rely on this – as well as the navigation there are links from components and patterns to other related components and patterns, etc. As an example, there's a link from the [error summary component](https://deploy-preview-247--govuk-design-system-preview.netlify.com/components/error-summary/) to 'error messages' (under 'How it works') which doesn't end in a trailing slash, and which is broken when offline.

I don't think 'enforcing' trailing slashes is really enforceable, and I don't think we can rely on regularly testing this functionality either – is there a way we can modify the service worker to resolve either way?

Reply:
This is certainly something that can be investigated further. Without a trailing slash the SW will assume `/about` is referring to `/about.html`. It looks for the trailing slash before appending the `directoryIndex` setting. There is a prettyUrl setting that didn't fix the issue. There is a [promising comment here](https://github.com/GoogleChrome/workbox/issues/903) that we can look at.
 
I think as @alex-ju mentioned above we should switch to the `injectManifest` method, this gives us a lot more fine grained control over how the SW works.

Reply:
This issue has now been fixed in the service worker without the need for modification.